presto-main/src/test/java/com/facebook/presto/sql/planner/TestLogicalPlanner.java

```diff
@@ -370,4 +373,17 @@ public void testCorrelatedScalarAggregationRewriteToLeftOuterJoin()
                                                                                 project(ImmutableMap.of("NON_NULL", expression("true")),
                                                                                         node(ValuesNode.class)))))))))));
     }
+
+    @Test
+    public void testPruneCountAggregationOverScalar()
```

Comment:
Also, I noticed that in LogicalPlanner.plan(), even before applying any optimizers, the AST looks to have been rewritten to an OutputNode followed by a series of ProjectNode children. Not sure when exactly this optimizer will see a tree containing an AggregationNode.